src/adapter/etl-adapter-google-sheet/src/Flow/ETL/Adapter/GoogleSheet/GoogleSheetLoader.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Flow\ETL\Adapter\GoogleSheet;
+
+use Flow\ETL\{Adapter\GoogleSheet\RowsNormalizer\EntryNormalizer, FlowContext, Loader, Row\Entry, Rows};
+use Google\Service\Sheets;
+use Google\Service\Sheets\ValueRange;
+
+final class GoogleSheetLoader implements Loader, Loader\Closure
+{
+    private ?int $columnsCount = null;
+
+    private string $dateFormat = 'Y-m-d';
+
+    private string $dateTimeFormat = \DateTimeInterface::ATOM;
+
+    private int $startingRow = 0;
+
+    private string $timeFormat = 'H:i';
+
+    private bool $withHeader = true;
+
+    public function __construct(
+        private readonly Sheets $service,
+        private readonly string $spreadsheetId,
+        private readonly string $sheetName,
+    ) {
+    }
+
+    public function closure(FlowContext $context) : void
+    {
+        $this->startingRow = 0;
+        $this->columnsCount = null;
+    }
+
+    public function load(Rows $rows, FlowContext $context) : void
+    {
+        if (!$rows->count()) {
+            return;
+        }
+
+        if (null === $this->columnsCount) {
+            $this->columnsCount = $rows->first()->entries()->count();
+        }
+
+        $currentRow = $this->calculateCurrentRow($rows);
+        $sheetRange = $this->calculateSheetRange($currentRow);
+
+        $normalizer = new RowsNormalizer(
+            new EntryNormalizer($context->config->caster(), $this->dateTimeFormat, $this->dateFormat, $this->timeFormat)
+        );
+
+        $this->service->spreadsheets_values->update(
+            $this->spreadsheetId,
+            $sheetRange->toString(),
+            new ValueRange(
+                [
+                    'values' => $this->values($rows, $normalizer),
+                    'majorDimension' => 'ROWS',
+                ]
+            ),
+            ['valueInputOption' => 'RAW']
```

Comment:
I would let end user to parametrize it, it's pretty convinient in many cases to let excel/google do the auto formatting: 

https://developers.google.com/workspace/sheets/api/reference/rest/v4/ValueInputOption?hl=pl

We just need an enum for that, you can literally copy description of each field to that enum to make it easier to understand

Reply:
During debug I removed the param, enum is there already 😂